src/lib.rs

```diff
@@ -29,6 +32,37 @@ pub struct Rect {
     pub height: u16
 }
 
+impl Rect {
+    /// Constructs new `Rect`.
+    pub fn new(left: u16, top: u16, width: u16, height: u16) -> Self {
+        Rect {
+            left: left,
+            top: top,
+            width: width,
+            height: height,
+        }
+    }
+
+    /// Constructs new zero-sized `Rect` placed at (0, 0).
+    pub fn new_empty() -> Self {
```

Comment:
Why should it be? It is jet another way of construction just like `new`. Compiler should optimize it so in binary code it will be just in-place initialization.